### PR TITLE
Fix Getting 500 error response for client errors in SCIM/Groups PATCH

### PR DESCRIPTION
**Pull request by PasinduYeshan**
## Purpose

This PR aims to address the issues mentioned in https://github.com/wso2/product-is/issues/16097.

## Goals

The goals of this PR are to:

- Change the error responses when adding a member to a group with invalid display name.
- Change the error responses when adding a member to a group with invalid id.
- Change the error responses when a user is added with mismatching id and display name.
- Change the error response when removing a non-existent user from a group
- Fix the issue of receiving a 400 error when attempting to `remove and add a user` scenario by using the user ID. However, this was working with the user display name.

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 1]
```diff
@@ -3366,6 +3367,11 @@ private void doPatchGroup(String groupId, String currentGroupName, Map<String, L
                 temporaryMembers.clear();
 
                 for (String member : deletedMembers) {
+                    if (addedMembers.isEmpty() && (member == null || member.isEmpty())) {
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:3370]: can't we use StringUtils here?

Comment (chamathns) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:3370]: let's refactor this.

eg:
```java
for (String member : deletedMembers) {
    if (addedMembers.isEmpty() && StringUtils.isBlank(member)) {
        throw new BadRequestException(ResponseCodeConstants.INVALID_VALUE);
    } 

    if (StringUtils.isNotBlank(member)) {
        String username = UserCoreUtil.addDomainToName(UserCoreUtil.removeDomainFromName(member), userStoreDomainForGroup);
        temporaryMembers.add(username);
    }
}
```

Comment (PasinduYeshan) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java]: Refactored in this [fedd887](https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/478/commits/fedd8873c8f5897f92c93a9d7f57cb4815026047)

File: .github/workflows/pr-builder.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
 
     strategy:
       matrix:
-        java-version: [ 11, 17 ]
+        java-version: [ 11.0.19+7 ]
```
Comment (chamathns) [inline .github/workflows/pr-builder.yml:22]: any reason for removing java version 17?

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java [hunk 3]
```diff
@@ -3413,13 +3419,16 @@ private void doPatchGroup(String groupId, String currentGroupName, Map<String, L
             carbonUM.updateGroupName(currentGroupName, newGroupName);
 
         } catch (UserStoreException e) {
+            if (e instanceof org.wso2.carbon.user.core.UserStoreException && StringUtils
+                    .equals(UserCoreErrorConstants.ErrorMessages.ERROR_CODE_NON_EXISTING_USER.getCode(),
+                            ((org.wso2.carbon.user.core.UserStoreException) e).getErrorCode())) {
+                throw new BadRequestException(ResponseCodeConstants.INVALID_VALUE);
```
Comment (chamathns) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:3425]: why aren't we passing the throwable?

File: .github/workflows/pr-builder.yml (hunk 2 above)
Comment (PasinduYeshan) [reply .github/workflows/pr-builder.yml:22]: I followed this https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/6cb4c30737e62537aad1d94c07e88283dcf80198/.github/workflows/pr-builder.yml#L23 as a workaround for the build failure.

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java (hunk 3 above)
Comment (PasinduYeshan) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:3425]:
```
public class BadRequestException extends AbstractCharonException {
    public BadRequestException() {
        this("invalid request");
    }

    public BadRequestException(String scimType) {
        this("Request is unparsable, syntactically incorrect, or violates schema.", scimType);
    }

    public BadRequestException(String details, String scimType) {
        super(400, details, scimType);
    }
}
```

This is the implementation of `BadRequestException`. There's no way to pass the throwable to the `BadRequestException`.

File: .github/workflows/pr-builder.yml (hunk 2 above)
Comment (chamathns) [reply .github/workflows/pr-builder.yml:22]: include [jdk-17.0.7+7](https://github.com/adoptium/temurin17-binaries/releases/tag/jdk-17.0.7%2B7) as well

Comment (PasinduYeshan) [reply .github/workflows/pr-builder.yml]: Added in [bea2a6a](https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/478/commits/bea2a6a4c22c549bbf6db771c25d1df238cd854a)

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java (hunk 3 above)
Comment (PasinduYeshan) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java:3426]: Added a log with the throwable as suggested offline.

**Review by chamathns — APPROVED** [review]

**Review by Kanapriya — APPROVED** [review]

**Pull request merged**